tests/e2e/k8s_test.go

```diff
@@ -172,3 +277,63 @@ func requireRunningPods(kubectlOptions *k8s.KubectlOptions, matchingLabelKey str
 		Expect(pod.Status.Phase).To(BeEquivalentTo("Running"))
 	}
 }
+
+// getK8sResources gets the specified K8S resources from the specified kubectl context and
+// namespace optionally. Extra arguments can be any of the kubectl get flag arguments.
+// Returns a slice of the returned elements. Separator between elements must be newline.
+func getK8sResources(kubectlOptions *k8s.KubectlOptions, resourceKind []string, selector string, names string, extraArgs ...string) []string {
+	logMsg := fmt.Sprintf("Get K8S resources: '%s'", resourceKind)
+
+	args := []string{"get", strings.Join(resourceKind, ",")}
+	logMsg, args = _kubectlArgExtender(args, logMsg, selector, names, kubectlOptions.Namespace, extraArgs)
+
+	By(logMsg)
+
+	args = append(args, extraArgs...)
+
+	output, err := k8s.RunKubectlAndGetOutputE(
+		GinkgoT(),
+		kubectlOptions,
+		args...,
+	)
+
+	Expect(err).NotTo(HaveOccurred())
+
+	output = strings.Trim(output, "'")
+	if output == "" {
+		return nil
+	}
+
+	output = strings.TrimRight(output, "\n")
+	outputSlice := strings.Split(output, "\n")
+	var resources []string
+
+	for i := range outputSlice {
+		if !strings.Contains(outputSlice[i], "Warning:") {
+			resources = append(resources, outputSlice[i])
+		}
+	}
+
+	return resources
+}
+
+func waitK8sResourceCondition(kubectlOptions *k8s.KubectlOptions, resourceKind, waitFor, timeout string, selector string, names string, extraArgs ...string) {
```

Comment:
Suggestion: Maybe we could use the go time package instead of a string for the timeout.

Reply:
It seems time.Duration can be used. I can change it but we need to change all of our timeout values.
I think it is a good idea. Im going to do this.

Reply:
done: commit [1fb27d3](https://github.com/banzaicloud/koperator/pull/987/commits/1fb27d30d3b0a1fb58329fb35d01233344f076cb)